File: 0100-same-tree/0100-same-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
       
        p_visited=[]
        q_visited=[]



        def bfs(p_node,q_node):
            if p_node is None and q_node is None:
                return True

            if p_node is None or q_node is None:
                return False

            
            q=deque()
            q.append((p_node,q_node))
           

            while q:
                cur_p, cur_q=q.popleft()
                
                p_visited.append(cur_q)
                q_visited.append(cur_p)

                if cur_p.val != cur_q.val:
                    return False

                if cur_p.left and cur_q.left:
                    q.append((cur_p.left, cur_q.left))
                elif cur_p.left or cur_q.left:
                    return False

                if cur_p.right and cur_q.right:
                    q.append((cur_p.right, cur_q.right))
                elif cur_p.right or cur_q.right:
                    return False

            return True 

        return bfs(p,q)

            



    

  

File: 0100-same-tree/0100-same-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} p
 * @param {TreeNode} q
 * @return {boolean}
 */
var isSameTree = function(p, q) {

    if(!p&&!q) return true
    if(!p || !q) return false

    const queue=[[p,q]]
    
    while(queue.length>0){
        const [pNode,qNode]=queue.shift()

        if(pNode==null && qNode==null) continue 
        if(pNode ==null || qNode ==null) return false
        if(pNode.val!=qNode.val) return false

        queue.push([pNode.left,qNode.left])
        queue.push([pNode.right,qNode.right])
    }
    
    return true
};